qemu/tests/dump_guest_core.py

```diff
@@ -0,0 +1,104 @@
+        logging.info("host kernel version is %s" % host_kernel_version)
+        logging.info("guest kernel version is %s" % guest_kernel_version)
+        if host_kernel_version != guest_kernel_version:
+            test.cancel("Please update your host and guest kernel "
+                        "to same version")
+
+    def check_core_file():
+        """
+        Use gdb to check core dump file
+        """
+        arch = 'X86_64'
+        arch = 'ppc64-le' if params['vm_arch_name'] == 'ppc64le' else arch
+        command = ('echo -e "source %s\nset height 0\ndump-guest-memory'
+                   ' %s %s\nbt\nquit" > %s' % (dump_guest_memory_file,
+                                               vmcore_file, arch,
+                                               gdb_command_file))
+        process.run(command, shell=True)
+        status, output = process.getstatusoutput(gdb_command, timeout=360)
+        os.remove(gdb_command_file)
+        os.remove(core_file)
+        logging.debug(output)
+        if status != 0:
+            test.fail("gdb command execute failed")
+        elif "<class 'gdb.MemoryError'>" in output:
+            if dump_guest_core == "on":
+                test.fail("Cannot access memory")
+
+    def check_vmcore_file():
+        """
+        Use crash to check vmcore file
+        """
+        process.run('echo -e "bt\ntask 0\ntask 1\nquit" > %s'
+                    % crash_script, shell=True)
+        crash_cmd = "crash -i %s /usr/lib/debug/lib/modules/%s/vmlinux %s"
+        crash_cmd %= (crash_script, host_kernel_version, vmcore_file)
+        status, output = process.getstatusoutput(crash_cmd, timeout=60)
+        os.remove(crash_script)
+        os.remove(vmcore_file)
+        logging.debug(output)
+        if "systemd" in output and 'swapper' in output:
+            logging.info("Crash command works as expected")
+        else:
+            test.fail("Vmcore corrupt")
+
+    # install crash/gdb/kernel-debuginfo in host
+    packages = ["crash", "gdb", "kernel-debuginfo*", "qemu-kvm-debuginfo",
+                "qemu-kvm-debugsource", "qemu-kvm-core-debuginfo"]
+    utils_package.package_install(packages)
+
+    trigger_core_dump_command = params["trigger_core_dump_command"]
+    dump_guest_memory_file = params["dump_guest_memory_file"]
+    gdb_command = params["gdb_command"]
+    gdb_command_file = params["gdb_command_file"]
+    dump_guest_core = params["dump_guest_core"]
+    crash_script = params["crash_script"]
+    vmcore_file = params["vmcore_file"]
+    env_process.process(test, params, env,
+                        env_process.preprocess_image,
+                        env_process.preprocess_vm)
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+    host_kernel_version = process.getoutput("uname -r").strip()
+    check_env()
+
+    qemu_id = vm.get_pid()
+    gdb_command %= qemu_id
+    core_file = '/var/core.' + str(qemu_id)
```

Comment:
This should be a fixed variable defined before `gdb_command`, who should reference it, or `/var/core.` should be separately defined before `pre_command`, where reference it too.

Reply:
Updated the code.Thanks